classes/CategoryHelper.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Contao Open Source CMS
+ *
+ * Copyright (c) 2017 Heimrich & Hannot GmbH
+ *
+ * @author  Rico Kaltofen <r.kaltofen@heimrich-hannot.de>
+ * @license http://www.gnu.org/licences/lgpl-3.0.html LGPL
+ */
+
+namespace NewsCategories;
+
+class CategoryHelper
+{
+
+    /**
+     * Prepare the category object
+     *
+     * @param NewsCategoryModel $objCategory
+     *
+     * @return object
+     */
+    public static function prepareCategory(NewsCategoryModel $objCategory)
+    {
+        $category                = (object) $objCategory->row();
+        $category->name          = $category->frontendTitle ?: $category->title;
+        $category->class         = 'category_' . $category->id . ($category->cssClass ? (' ' . $category->cssClass) : '');
+        $category->linkTitle     = specialchars($category->frontendTitle ?: $category->title);
+        $category->href          = '';
+        $category->hrefWithParam = '';
+        $category->targetPage    = null;
+
+        // Add the target page
+        if (($targetPage = $objCategory->getTargetPage()) !== null)
+        {
+            $category->href          = $targetPage->getFrontendUrl();
+            $category->hrefWithParam = $targetPage->getFrontendUrl('/' . NewsCategories::getParameterName() . '/' . $category->alias);
+            $category->targetPage    = $targetPage;
+        }
+
+        // Register a function to generate category URL manually
+        $category->getUrl = function (\PageModel $page) use ($category) {
+            return $category->getUrl($page);
```

Comment:
This is just merged from your implementation in `NewsCategories\News::addCategoriesToTemplate Line 74`:

```
// Register a function to generate category URL manually
                        $arrCategories[$objCategory->id]['getUrl'] = function(\PageModel $page) use ($objCategory) {
                            return $objCategory->getUrl($page);
                        };
```